src/components/markdown_editor/plugins/markdown_link_validator.tsx

```diff
@@ -30,7 +30,15 @@ export function markdownLinkValidator() {
 
 export function mutateLinkToText(node: LinkOrTextNode) {
   node.type = 'text';
-  node.value = `[${node.children![0]?.value || ''}](${node.url})`;
+
+  // https://github.com/elastic/eui/issues/5770
+  // if this is a `mailto:` link only keep the target address
+  if (node.url?.startsWith('mailto:')) {
+    node.value = node.children![0]?.value;
+  } else {
+    node.value = `[${node.children![0]?.value || ''}](${node.url})`;
+  }
```

Comment:
minor readability suggestion:

```suggestion
  const linkText = node.children?.[0]?.value || '';

  if (node.url?.startsWith('mailto:')) {
    node.value = linkText;
  } else {
    node.value = `[${linkText}](${node.url})`;
  }
```

Reply:
👍 like that a lot